eds/interfaces/plugin_interface.py

```diff
@@ -1,23 +1,25 @@
-
 from __future__ import annotations
 from typing import List, Dict
+from abc import ABCMeta, abstractmethod
 
 
-class Plugin():
+class PluginInterface(metaclass=ABCMeta):
```

Comment:
I'd rather keep the interface naming consistent.   just simple meaningful names.   no "Interface" in there.  to remove the name conflict with the `Plugin` base class, let's rename that to `BasePlugin`.   see https://github.com/manheim/eds/pull/25

Reply:
I agree that the naming should be changed, however I don't really like `BasePlugin`. From my understanding the `PluginInterface` is the "Abstract Base" class.  So `Plugin` would be an implementation of plugin interface, therefore not a base class.

Reply:
I followed the convention mentioned in this stack overflow: https://stackoverflow.com/a/2814831

I've seen that pattern before.

*  the interface is named for what it is
*  Using "Base.." or "Default.."  for abstract shared implementation classes

In our case, `PluginInterface` is intended as an "interface", not technically an "abstract" base class.

```
An abstract class permits you to make functionality that subclasses can implement or override whereas an interface only permits you to state functionality but not to implement it"
```

I agree that using "ABC" makes things somewhat awkward due to it's name, but I like following he broader standard.